core/test/usFrameworkTest.cpp

```diff
@@ -113,30 +134,43 @@ namespace
     void TestLifeCycle()
     {
         TestBundleListener listener;
-        FrameworkFactory factory;
         std::vector<BundleEvent> pEvts;
 
-        std::map < std::string, Any > configuration;
-        //configuration[Constants::FRAMEWORK_LOG_LEVEL] = 0;
-        auto f = factory.NewFramework(configuration);
+        auto f = FrameworkFactory().NewFramework();
 
         US_TEST_CONDITION(f.GetState() == Bundle::STATE_INSTALLED, "Check framework is installed")
 
+        // make sure WaitForStop returns immediately when the Framework's state is "Installed"
+        auto fNoWaitEvent = f.WaitForStop(std::chrono::milliseconds::zero());
+        US_TEST_CONDITION(fNoWaitEvent, "Check for valid framework event");
+        US_TEST_CONDITION(fNoWaitEvent.GetType() == FrameworkEvent::Type::FRAMEWORK_ERROR, "Check for correct framework event type");
+        US_TEST_CONDITION(fNoWaitEvent.GetThrowable() == nullptr, "Check that no exception was thrown");
+
         f.Init();
 
-        US_TEST_CONDITION(f.GetState() == Bundle::STATE_STARTING, "Check framework is starting")
+        US_TEST_CONDITION(f.GetState() == Bundle::STATE_STARTING, "Check framework is starting");
+        US_TEST_CONDITION(f.GetBundleContext(), "Check for a valid bundle context");
 
         f.Start();
 
-        US_TEST_CONDITION(listener.CheckListenerEvents(pEvts), "Check framework bundle event listener")
+        US_TEST_CONDITION(listener.CheckListenerEvents(pEvts), "Check framework bundle event listener");
 
-        US_TEST_CONDITION(f.GetState() == Bundle::STATE_ACTIVE, "Check framework is active")
+        US_TEST_CONDITION(f.GetState() == Bundle::STATE_ACTIVE, "Check framework is active");
 
         f.GetBundleContext().AddBundleListener(&listener, &TestBundleListener::BundleChanged);
 
+        // To test that the correct FrameworkEvent is returned, we must guarantee that the Framework
+        // is in a STARTING, ACTIVE, or STOPPING state.
+        std::thread waitForStopThread([&f]() {
```

Comment:
For gcc, a `#include <thread>` statement is now required due to the usage of `std::thread`.

Also, we should "ifdef" that such that it is only executed in the multi-thread configuration.